### PR TITLE
Feature/#41 make required fields optional for basic api support

### PR DESCRIPTION
**Pull request by saa14**
- Removed required validations on Namespace and Type from Edit and Create flows and modified MappingProfile to have an empty string in place of null for them.
- Refactored setSaveButton with disableSaveButton and added a similar function on Create to disable the Save button on the page when all rows are deleted from the root grid. 
- Removed the locks on timestamp and id. 
- Removed checks for null for id and timestamp.Added a method updateSaveButton and added a setSaveButton on Create.cshtml to control save button state on both Edit and Create. This also takes care of the case when the root grid is empty.
- Refactored tests (backend and ui) to reflect the changes in validations

### CODE REVIEW

File: src/Totem.Tests/Features/Contracts/CreateTests.cs [hunk 1]
```diff
@@ -243,77 +241,6 @@ public void ShouldNotCreateWhenContractStringIsNotValidSchema()
             command.ShouldNotValidate("Contract must be defined as a valid OpenAPI schema.");
         }
```
Comment (najeraz) [inline src/Totem.Tests/Features/Contracts/CreateTests.cs]: SHOULD: Add integration test(s) for creating a contract without Id, Timestamp, Type (null), and Namespace (null)

File: src/Totem.Tests/Features/Contracts/EditTests.cs [hunk 2]
```diff
@@ -503,59 +447,6 @@ public async Task ShouldNotEditWhenContractStringTimestampDoesNotHaveFormat()
             command.ShouldNotValidate("The Timestamp property must have a format of date-time.");
         }
```
Comment (najeraz) [inline src/Totem.Tests/Features/Contracts/EditTests.cs]: SHOULD: Add integration test(s) for editing a contract removing Id, Timestamp, Type (null), and Namespace (null).

File: src/Totem/ClientApp/features/contracts/__tests__/e2e/RootLevelSingleField.e2e.js [hunk 3]
```diff
@@ -180,6 +181,39 @@ test('Deleting a previously saved root field', async t => {
...
+
+  await t.click(
+    Selector('tr.treegrid-body-row')
+      .nth(0)
+      .find('.edit-action')
+  );
+  await t.click(utils.deleteFieldBtn);
+
+  await t.expect(Selector('tr.treegrid-body-row').count).eql(0);
+  await t.expect(Selector('#rootGrid .treegrid-empty-row').count).eql(1);
```
[the remark this answers is not in the document]
Comment (saa14) [reply src/Totem/ClientApp/features/contracts/__tests__/e2e/RootLevelSingleField.e2e.js]: The saveButton disable check works on manual testing but does not while testcafe runs the test. The functions seem to execute properly but there is a webpack import error on testcafe which halts the disable code. Will need to work around that.

File: src/Totem.Tests/Features/Contracts/CreateTests.cs (hunk 1 above)
Comment (saa14) [reply src/Totem.Tests/Features/Contracts/CreateTests.cs]: Updated with f36a51801c338d9897cf50b74466bea3817029e8

File: src/Totem.Tests/Features/Contracts/EditTests.cs (hunk 2 above)
Comment (saa14) [reply src/Totem.Tests/Features/Contracts/EditTests.cs]: Updated with f36a51801c338d9897cf50b74466bea3817029e8

**Comment by najeraz** [comment]
@saa14 Still two e2e tests failing.
